crates/qt-build-utils/src/lib.rs

```diff
@@ -306,8 +306,84 @@ impl QtBuild {
         .to_string()
     }
 
+    fn cargo_link_qt_library(
+        &self,
+        name: &str,
+        prefix_path: &str,
+        lib_path: &str,
+        link_lib: &str,
+        prl_path: &str,
+        builder: &mut cc::Build,
+    ) {
+        println!("cargo:rustc-link-lib={link_lib}");
+
+        match std::fs::read_to_string(prl_path) {
+            Ok(prl) => {
+                for line in prl.lines() {
+                    if let Some(line) = line.strip_prefix("QMAKE_PRL_LIBS = ") {
+                        parse_cflags::parse_libs_cflags(
+                            name,
+                            line.replace(r"$$[QT_INSTALL_LIBS]", lib_path)
+                                .replace(r"$$[QT_INSTALL_PREFIX]", prefix_path)
+                                .as_bytes(),
+                            builder,
+                        );
+                    }
+                }
+            }
+            Err(e) => {
+                println!(
+                    "cargo:warning=Could not open {} file to read libraries to link: {}",
+                    &prl_path, e
+                );
+            }
+        }
+    }
+
+    /// Some prl files don't follow a consistent naming scheme. Try to find it by looking at all files in lib_path.
+    /// The android libraries adds its architecture at the end of it prl files eg liQt6Core_{arch}.prl.
+    /// The arch placeholder is somewhat different for each architecture preventing the use of a simple format.
+    fn find_qt_module_prl(
+        &self,
+        lib_path: &str,
+        prefix: &str,
+        version_major: u32,
+        qt_module: &str,
+    ) -> String {
+        match Path::new(lib_path).read_dir() {
+            Ok(lib_dir) => {
+                for entry in lib_dir {
+                    match entry {
+                        Ok(entry) => {
+                            let file_name = entry.file_name();
+                            let file_name = file_name.to_string_lossy();
+                            let prl_file = file_name.ends_with(".prl");
+                            let matches_module = file_name
+                                .contains(&format!("Qt{}{}", self.version.major, qt_module));
```

Comment:
This isn't specific enough. It finds libQt6QuickDialogs2Utils.prl when linking Qt6Quick which doesn't list all the `.o` files that Qt6Quick needs to link. Can you post a list of those filenames for Android? Maybe we can hardcode a list of Android architectures to append or something?